contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -0,0 +1,102 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "../../interfaces/INFT.sol";
+import "../../interfaces/IStreamRegistry.sol";
+import "../../libraries/nodes/VehicleStorage.sol";
+import "../../libraries/streamr/StreamrManagerStorage.sol";
+import "../../libraries/streamr/VehicleStreamStorage.sol";
+
+import "../../shared/Errors.sol" as Errors;
+
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/// TODO Documentation
+contract VehicleStream is AccessControlInternal {
+    string constant DIMO_STREAM_ENS = "streams.dimo.eth";
+    string constant DIMO_STREAM_ENS_VEHICLE = "streams.dimo.eth/vehicle/";
+
+    // TODO Documentation
+    function createStream(uint256 vehicleId) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        IStreamRegistry streamRegistry = IStreamRegistry(
+            StreamrManagerStorage.getStorage().streamrRegistry
+        );
+
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        string memory streamPath = string(
+            abi.encodePacked("/vehicle/", Strings.toString(vehicleId))
```

Comment:
Why not use `DIMO_STREAM_ENS_VEHICLE`? I wonder if this should be configurable. It feels weird to be a constant.

Reply:
This `streamPath` must not contain the ENS. The ENS is a separate param in the streamr function.

You mean configurable the ENS or the path? I had the ENS in the storage first, but I don't think it will change, or might change... not sure

Reply:
I'm dumb.